socorro/external/boto/crashstorage.py

```diff
@@ -28,13 +30,101 @@ def wait_time_generator():
         yield i
 
 
+class CrashidMissingDatestamp(Exception):
+    """Indicates the crash id is invalid and missing a datestamp."""
+
+    pass
+
+
+def get_datestamp(crashid):
+    """Parses out datestamp from a crashid.
+
+    :raises CrashidMissingDatestamp: if the crash id has no datestamp at the end
+
+    """
+    datestamp = date_from_ooid(crashid)
+    if datestamp is None:
+        # We should never hit this situation unless the crashid is not valid
+        raise CrashidMissingDatestamp("%s is missing datestamp" % crashid)
+    return datestamp
+
+
+def build_keys(name_of_thing, crashid):
```

Comment:
This is still titled `build_keys` even though it returns a list of a single key in all cases. I think it's likely we'll want to do a migration pass removing the entropy from the paths and figure it's easier to leave this as is.

If, however, you think that's not worth waiting for, I can spend some time de-pluralizing this in this PR.